client/consensus/pow/src/lib.rs

```diff
@@ -391,7 +399,24 @@ impl<B, I, C, S, Algorithm, CAW> BlockImport<B> for PowBlockImport<B, I, C, S, A
 		block.auxiliary.push((key, Some(aux.encode())));
 		if block.fork_choice.is_none() {
 			block.fork_choice = Some(ForkChoiceStrategy::Custom(
-				aux.total_difficulty > best_aux.total_difficulty
+				match aux.total_difficulty.cmp(&best_aux.total_difficulty) {
+					Ordering::Less => false,
+					Ordering::Greater => true,
+					Ordering::Equal => {
+						let best_inner_seal = match best_header.digest().logs.last() {
+							Some(DigestItem::Seal(id, seal)) => {
+								if id == &POW_ENGINE_ID {
+									seal.clone()
+								} else {
+									return Err(Error::<B>::WrongEngine(*id).into())
+								}
+							},
+							_ => return Err(Error::<B>::HeaderUnsealed(best_hash).into()),
+						};
```

Comment:
You have the almost exact same code some lines above. You could put this into a function :D